pylint/checkers/refactoring.py

```diff
@@ -27,6 +27,52 @@ def _all_elements_are_true(gen):
     return values and all(values)
 
 
+def _get_exception_handlers(node, exception):
+    """Return the collections of handlers handling the exception in arguments"""
+    current = node
+    ignores = (astroid.ExceptHandler, astroid.TryExcept)
+    while current and not isinstance(current.parent, ignores):
+        current = current.parent
+
+    if current and isinstance(current.parent, astroid.TryExcept):
+        return [_handler for _handler in current.parent.handlers if _handler.catch(exception)]
+    return None
+
+
+def _is_node_return_ended(node):
+    """Return True if the node ends with an explicit return statement"""
+    # Recursion base case
+    if isinstance(node, astroid.Return):
+        return True
+    if isinstance(node, astroid.Raise):
+        # a Raise statement doesn't need to end with a return statement
+        # but if the exception raised is handled, then the handler has to
+        # ends with a return statement
+        exc = utils.safe_infer(node.exc)
+        exc_name = exc.pytype().split('.')[-1]
+        if exc is None:
+            return False
+        handlers = _get_exception_handlers(node, exc_name)
+        if handlers:
+            # among all the handlers handling the exception at least one
+            # must end with a return statement
+            return any(_is_node_return_ended(_handler) for _handler in handlers)
+        # if no handlers handle the exception then it's ok
+        return True
+    if isinstance(node, astroid.If):
+        # if statement is returning if there are exactly two return statements in its
+        # children : one for the body part, the other for the orelse part
+        return_stmts = [_is_node_return_ended(_child) for _child in node.get_children()]
+        return_stmts = [_rs for _rs in return_stmts if _rs]
```

Comment:
Trivial: You could simplify this slightly by removing this line and using `return sum(return_stmts) == 2`. Or instead keep this line and make the line below `return len(return_stmts) == 2`.